src/shared.ts

```diff
@@ -72,14 +72,21 @@ export const hashedString = (string: string) => {
     return hash;
 };
 
-export const getSchemas = ({ json, swaggerVersion = 3 }: GetSchemasProps) => {
-    switch (swaggerVersion) {
-        case 3:
-            return json?.components?.schemas;
-        case 2:
-            return json?.definitions;
-        default:
-            return json?.components?.schemas;
+export function isSwaggerV3(json: SwaggerV2 | SwaggerV3): json is SwaggerV3 {
+    return Boolean((json as SwaggerV3).components) && (json as SwaggerV3).openapi === '3.0.0';
```

Comment:
What if version is later `3.1.0` would the code break?
Should we instead check if it starts with `3.`
or use regex?
`openapi.match(/^3.*/)` ?

Reply:
@ardeois I thought about it, but... I have not tested it for other version other than 2.0 and 3.0. I have no guarantee that it will not fail, but I can remove that restriction, sure, np.

Reply:
Yes I would prefer to remove the restriction. Otherwise it will just crash if the version is not known
So it's better to try to parse 3.x as 3.0 and fix it later if there is breaking changes